lib/spack/llnl/util/symlink.py

```diff
@@ -2,77 +2,166 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
-import errno
 import os
 import shutil
 import sys
 import tempfile
-from os.path import exists, join
 
-from llnl.util import lang
+from llnl.util import lang, tty
+
+from spack.error import SpackError
+from spack.util.path import system_path_filter
 
 if sys.platform == "win32":
     from win32file import CreateHardLink
 
 
-def symlink(real_path, link_path):
+def symlink(source_path: str, link_path: str):
     """
-    Create a symbolic link.
+    Try to create a symbolic link.
```

Comment:
Rewording suggestion: remove the word "try" - we either do it or we raise an exception.